pkg/ccr/base/spec.go

```diff
@@ -622,6 +622,12 @@ func (s *Spec) CreateTableOrView(createTable *record.CreateTable, srcDatabase st
 
 	log.Infof("create table or view sql: %s", createSql)
 
+	// 'COMMENT '[xxx, xxx]' should be "COMMENT '[xxx, xxx]"
+	re := regexp.MustCompile(`COMMENT\s+'\[.*?\]'`)
+	createSql = re.ReplaceAllStringFunc(createSql, func(match string) string {
+		return `COMMENT "` + match[len("COMMENT '"):len(match)-1] + `"`
```

Comment:
The right way is escaping the single quote, in the matched comments.

BTW, instead of extracting content by `len`, consider using the capturing groups, to extract the comment content directly.